File: java/VIPERArchi1/app/src/main/java/com/singularitycoder/viperarchi1/ApiEndPoints.java
package com.singularitycoder.viperarchi1;

import com.singularitycoder.viperarchi1.NewsHome.Entity.NewsItemResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiEndPoints {

    @GET("top-headlines")
    Call<NewsItemResponse> getNewsList(
            @Query("country") String country,
            @Query("category") String category,
            @Query("apiKey") String apiKey
    );
}

File: kotlin/DaggerHilt1/app/src/main/java/com/singularitycoder/daggerhilt1/ApiEndPoints.kt
package com.singularitycoder.daggerhilt1

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiEndPoints {

    @GET("/search/repositories")
    fun searchAllRepositories(
        @Query("q") query: String,
        @Query("sort") sort: String,
        @Query("order") order: String,
    ): Call<RepoResponse>
}